File: 110.平衡二叉树.cpp
/*
 * @lc app=leetcode.cn id=110 lang=cpp
 *
 * [110] 平衡二叉树
 */

// @lc code=start
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution
{
public:
    bool flag = true; 
    bool isBalanced(TreeNode *root)
    {
        depth(root);
        return flag;
    }
    int depth(TreeNode *root)
    {
        if (root != nullptr)
        {
            int l =depth(root->left);
            int r =depth(root->right);
            if(abs(l - r) > 1)
            {
                flag = false;
            }
            return max(l,r) + 1;
        }
        return 0;
        
    }
};
// @lc code=end

File: 110.平衡二叉树.java
/*
 * @lc app=leetcode.cn id=110 lang=java
 *
 * [110] 平衡二叉树
 */

// @lc code=start
/**
 * Definition for a binary tree node. public class TreeNode { int val; TreeNode
 * left; TreeNode right; TreeNode() {} TreeNode(int val) { this.val = val; }
 * TreeNode(int val, TreeNode left, TreeNode right) { this.val = val; this.left
 * = left; this.right = right; } }
 */
class Solution {
    public boolean isBalanced(TreeNode root) {
        if (root == null)
            return true;
        return Math.abs(length(root.left) - length(root.right)) < 2 && isBalanced(root.left) && isBalanced(root.right);
    }

    public int length(TreeNode node) {
        if (node == null) {
            return 0;
        }
        return Math.max(length(node.left), length(node.right)) + 1;
    }
}
// @lc code=end
